packages/langchain/src/openai/util.ts

```diff
@@ -232,8 +238,10 @@ function mapBaseMessageToAzureOpenAiChatMessage(
 }
 
 function isStructuredToolArray(tools?: unknown[]): tools is StructuredTool[] {
```

Comment:
This function would have to be changed if we switch to using `BindToolsInput` instead.
LangChain already offers a utility function `isLangChainTool` to check if it the tool is in LangChain structured tool format.

Reply:
I decided to implement our own `isToolDefinition()` to check the other half. See comments in the code.